### PR TITLE
add RHUI extras repo for *-devel packages

### PR DESCRIPTION
**Pull request by neilmillard-msm**

### CODE REVIEW

**Comment by gregbrown-msm** [comment]
:+1:

**Comment by neilmillard-msm** [comment]
check code execution on CentOS7
may need more specific conditional

File: init.sh [hunk 1]
```diff
@@ -153,6 +154,18 @@ set_facter() {
   cat /etc/facter/facts.d/"${1}".txt
 }
 
+setup_rhel7_repo() {
+  yum install -y redhat-lsb-core
```
Comment (seanclerkin) [inline init.sh]: We have a helper function yum_install which is faster if the package is already installed

File: init.sh [hunk 2]
```diff
@@ -153,6 +154,18 @@ set_facter() {
...
+  dist=$(lsb_release -is)
+  majorversion=$(lsb_release -rs | cut -f1 -d.)
+  if [[ "$majorversion" == "7" ]] & [[ "$dist" == "RedHatEnterpriseServer" ]]; then
+    echo "RedHat Enterprise version 7- adding extra repo for *-devel"
+    yum install -y yum-utils
```
Comment (seanclerkin) [inline init.sh]: We have a helper function yum_install which is faster if the package is already installed

File: init.sh [hunk 3]
```diff
@@ -153,6 +154,18 @@ set_facter() {
...
+    yum-config-manager --enable rhui-REGION-rhel-server-optional
+    yum install -y ruby-devel
```
Comment (seanclerkin) [inline init.sh]: We have a helper function yum_install which is faster if the package is already installed

**Comment by seanclerkin** [comment]
:+1:

File: init.sh [hunk 4]
```diff
@@ -153,6 +154,18 @@ set_facter() {
   cat /etc/facter/facts.d/"${1}".txt
 }
 
+setup_rhel7_repo() {
+  yum_install redhat-lsb-core
```
Comment (TristanPeers) [inline init.sh]: Do you not need a -y here?

**Comment by TristanPeers** [comment]
:+1:

File: init.sh [hunk 5]
```diff
@@ -153,6 +154,18 @@ set_facter() {
...
+  dist=$(lsb_release -is)
+  majorversion=$(lsb_release -rs | cut -f1 -d.)
+  if [[ "$majorversion" == "7" ]] & [[ "$dist" == "RedHatEnterpriseServer" ]]; then
```
Comment (dstockton) [inline init.sh]: Stylistically, I prefer...
```
if [ "$majorversion" == "7" ] && [ "$dist" == "RedHatEnterpriseServer" ]; then
```

but :+1:

**Pull request merged**